### PR TITLE
Add black

### LINKED ISSUE #200: Consider using black or yapf for Python code formatting
**Is your feature request related to a problem? Please describe.**
Based on the premise of having an unified code style on projects, it'd be great if we add support to a code formatting tool.

**Describe the solution you'd like**
This can be done either during development by the developers themselves or integrated as a pre-commit hook.

**Describe alternatives you've considered**
There are two popular tools to achieve this:
- [google/yapf](https://github.com/google/yapf)
- [ambv/black](https://github.com/ambv/black)

**Additional context**
At a first glance, seems like the latter is not configurable, while the former is. IMHO, this should be taken into consideration.

### PR DESCRIPTION
**Pull request by vanessa**
## Description

* Add black to project (along with configs in `pyproject.toml`)
* Format files in backend folder
* Add black to pre-commit hook
* Fix has_missing_migrations on pre-commit hook configs
* Add generated requirements files (.txt) to `.gitignore`

> Note: Ignoring `.py` files in "Files changed" section removes the formatted files and cleans the diff a little bit, [here's a shortcut](https://github.com/vintasoftware/django-react-boilerplate/pull/436/files?file-filters%5B%5D=.in&file-filters%5B%5D=.md&file-filters%5B%5D=.toml&file-filters%5B%5D=.yaml&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile)

## Motivation and Context

Closes #200 

### black

1. Clone the project as a user following [these steps](https://github.com/vintasoftware/django-react-boilerplate/blob/master/CONTRIBUTING.md#testing-your-changes)
1. After you compile and install requirements, check if black is installed by typing `black`
	* [ ] The output is `No Path provided. Nothing to do 😴`
1. Install pre-commit hook with `pre-commit install`
1. Create an example Python file inside `backend` folder
	* Use the example in [black playground](https://black.now.sh/)
1. Add the same file created previously inside a migrations folder (`users/migrations` for example)
1. Try to commit the file you created (or run `pre-commit`)
	* [ ] pre-commit hook runs and formats the file
	* [ ] The limit of 100 characters per line is respected
	* [ ] The file in migrations folder wasn't touched

### has_missing_migrations on pre-commit

1. Run pre-commit
	- [ ] `missing-migrations-local` runs without problems

### generated requirements files

1. Try to commit when you finish following steps in black section
	- [ ] The generated `requirements.txt` and `dev-requirements.txt` isn't included

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires documentation updates.
- [x] I have updated the documentation accordingly.
- [x] My change requires dependencies updates.
- [x] I have updated the dependencies accordingly.

### CODE REVIEW

**Comment by chocoelho** [comment]
Please, merge `isort.cfg` with the following configuration as suggested by Black, remove the duplicates and make the suggestions by Black supersed ours.:

```
[settings]
multi_line_output=3
include_trailing_comma=True
force_grid_wrap=0
use_parentheses=True
```

**Comment by vanessa** [comment]
@chocoelho Added them if you want to check :+1:

File: .gitignore [hunk 1]
```diff
@@ -47,3 +44,6 @@ spritesmith.scss
 
 # templated email
 tmp_email/
+
+# requirements
+*requirements.txt
```
Comment (vanessa) [inline .gitignore]: @jhgv Should I remove this?

Comment (jhgv) [reply .gitignore]: I think it is ok to let it here.

**Pull request merged**